pyanaconda/modules/network/network.py

```diff
@@ -652,20 +652,20 @@ def get_dracut_arguments(self, iface, target_ip, hostname, ibft):
         ))
         return dracut_args
 
-    def _apply_boot_options(self, kernel_arguments):
+    def _apply_boot_options(self, kernel_args):
         """Apply boot options to the module.
 
-        :param kernel_arguments: structure holding installer boot options
-        :type kernel_arguments: KernelArguments
+        :param kernel_args: structure holding installer boot options
+        :type kernel_args: KernelArguments
         """
-        log.debug("Applying boot options %s", kernel_arguments)
-        if 'ksdevice' in kernel_arguments:
-            self.default_device_specification = kernel_arguments.get('ksdevice')
-        if 'BOOTIF' in kernel_arguments:
-            self.bootif = kernel_arguments['BOOTIF'][3:].replace("-", ":").upper()
-        if 'ifname' in kernel_arguments:
-            self.ifname_option_values = kernel_arguments.get("ifname", "").split()
-        if 'noipv6' in kernel_arguments:
+        log.debug("Applying boot options %s", kernel_args)
+        if 'ksdevice' in kernel_args:
+            self.default_device_specification = kernel_args.get('ksdevice')
+        if 'BOOTIF' in kernel_args:
+            self.bootif = kernel_args['BOOTIF'][3:].replace("-", ":").upper()
```

Comment:
I think that this will not work.

Reply:
Thank you! Agreed. Makes me wonder two things:
- Is there some method you use to find these issues? I must be missing _something_, as I searched for all uses, and then got some more help from pylint, and then tests passed, and all of that was still not enough.
- Does anyone see any value in keeping the "I am a dictionary" behavior? I'd rather keep the interface small with just `get()`. But there is the question of addon usage, even if so far I did not find anyone using `flags.cmdline`...

Reply:
> Is there some method you use to find these issues?

Unfortunately, no. I have noticed it during reading your pull request. That is the reason why the reviews are so valuable. It is very common to overlook something during the implementation, so it is good to have a backup ;-)

>  Does anyone see any value in keeping the "I am a dictionary" behavior?

I think that the methods that we have now are sufficient. Btw. it looks like only the kdump addon uses `flags.cmdline`. The link to the upstream is https://github.com/daveyoung/kdump-anaconda-addon.

Reply:
I checked both kdump and oscap and I was unable to find any place where cmdline would be referenced, same as KernelArguments. I think that means case closed for them? Kdump only writes bootloader arguments, which is a different thing.

Reply:
https://github.com/daveyoung/kdump-anaconda-addon/blob/master/com_redhat_kdump/gui/spokes/kdump.py#L54

Reply:
Argh, it's a forked repo! I see where I made the mistake.